docs/architecture/components/2_payment_service_contract.md

```diff
@@ -0,0 +1,52 @@
+# Payment Service
+
+The Payment Service is the Component in charge of handling user's payments to fund the verification of their proofs. 
+
+To be able to use the batcher, a user must fund its transactions. For this, there is a simple Batcher Payment System.
+
+The Batcher has a Batcher Payments smart contract, which is in charge of recieving user's payments, and which guarantees the batcher can only spend this money to send users' proofs to Aligned.
+
+Users must first deposit into this contract, via a normal transfer to its address.
+
+Then, users can send proofs to the Batcher, the Batcher will preemptively check if the user has funds for this, and once accumulating the whole batch, the Batcher will call its smart contract with the data it has recieved from the users.
+
+The smart contract will then discount the corresponding amount of funds from each of the senders' balances, and create a new Batch in [Aligned Service Manager](./3_service_manager_contract.md), sending with it the corresponding amount of tokens for the batch verification.
+
+Users can then withdraw extra funds deposited to the Batcher Payments smart contract, or leave them to fund future proofs.
+
+This way, the Batcher can only use User funds to fund the verification of the User's proofs.
```

Comment:
This way, the Batcher can only use user funds to pay for the verification of the user's proofs.